ui-tests/src/test/java/org/openmrs/reference/AddDiagnosisToVisitNoteTest.java

```diff
@@ -36,7 +35,7 @@ public void AddDiagnosisToVisitNoteTest() throws Exception {
         assertEquals("Pneumonia", visitNotePage.primaryDiagnosis());
         assertEquals("Bleeding", visitNotePage.secondaryDiagnosis());
         visitNotePage.save();
-        assertNotNull(patientDashboardPage.visitLink());
+        visitNotePage.waitForElement(ClinicianFacingPatientDashboardPage.VISIT_LINK);
```

Comment:
visitNotePage.save() should return ClinicianFacingPatientDahboardPage and only ther you can look for elements. But looking for a toast message is bad. Look for entered diagnoses in the diagnoses section.